src/bytes.rs

```diff
@@ -66,22 +67,40 @@ use std::collections::HashMap;
 
 use crate::{Symbol, SymbolOverflowError, DEFAULT_SYMBOL_TABLE_CAPACITY};
 
-/// Wrapper around `&'static [u8]` that supports deallocating references created
-/// via [`Box::leak`].
+/// Wrapper around `&'static [u8]`.
 ///
 /// # Safety
 ///
 /// Must not be `Clone` or `Copy` because the Drop logic assumes this enum is the
```

Comment:
`Slice` actually can be `Clone` now, since `Box<_>` conveys unique ownership, and a clone of a `Box` creates a new heap object without touching the existing one.

The safety consideration here is rather that even though `Box` is a "unique owner" of the data, it should not be mutably dereferenced, because `as_static_slice` promises the slice to be valid as long as the `Slice` is not dropped.

<details><summary>Extra pedantic language-lawyering</summary>

(If you want to be _extra_ paranoid, you might require that `Slice` is not moved after `as_static_slice` is called. I think it's highly unlikely that it would be required to not move the `Box` while holding a reference to its contents, but depending on the exact formulation of the borrowing model for Rust that _could_ potentially be a valid interpretation of the status quo. (The `StableDeref` trait relies on this not being true, as does other existing unsafe code in the wild. The benefit of this requirement would be always being able to tag a passed-by-value `Box` pointer with LLVM `@unique`, rather than only once it has been mutably dereferenced.) (The API provided here is still possible to implement soundly even if that restriction is required in the future and not upholdable by the library, it just requires storing `*const _` rather than `Box<_>` to remove the extra information.))

Reply:
Thanks for the extra pedantic explanation. I think the pedantic version is easy enough to enforce by making an opaque wrapper around the Slice enum and using module visibility tricks.

Reply:
the result of this train of thought is the `internal` module. `Slice` is generic over the slice type and is wrapped by the `Interned` struct which disallows mutable access to the internals of `Slice`.